src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_AwaitCompletion.vb

```diff
@@ -386,6 +424,319 @@ public class C
+                state.SendTypeChars("aw")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True, inlineDescription:=FeaturesResources.Make_containing_scope_async)
+
+                state.SendTab()
+                Assert.Equal(GetTestClassDocument(containerHasAsyncModifier:=True, committed).Value.NormalizeLineEndings(), state.GetDocumentText().NormalizeLineEndings())
+                Await state.AssertLineTextAroundCaret($"        {committed}", "")
+            End Using
+
+            ' ConfigureAwait(false) starts here
+            committed += ".ConfigureAwait(false)"
+            ' place await in front of expression and append ConfigureAwait(false)
+            Using state = TestStateFactory.CreateCSharpTestState(GetTestClassDocument(containerHasAsyncModifier:=True, expression))
+                state.SendTypeChars("af")
+                Await state.AssertSelectedCompletionItem(displayText:="awaitf", isHardSelected:=True)
+
+                state.SendTab()
+                Assert.Equal(GetTestClassDocument(containerHasAsyncModifier:=True, committed).Value.NormalizeLineEndings(), state.GetDocumentText().NormalizeLineEndings())
+                Await state.AssertLineTextAroundCaret($"        {committed}", "")
+            End Using
+
+            ' place await in front of expression, append ConfigureAwait(false) and make container async
+            Using state = TestStateFactory.CreateCSharpTestState(GetTestClassDocument(containerHasAsyncModifier:=False, expression))
+                state.SendTypeChars("af")
+                Await state.AssertSelectedCompletionItem(displayText:="awaitf", isHardSelected:=True, inlineDescription:=FeaturesResources.Make_containing_scope_async)
+
+                state.SendTab()
+                Assert.Equal(GetTestClassDocument(containerHasAsyncModifier:=True, committed).Value.NormalizeLineEndings(), state.GetDocumentText().NormalizeLineEndings())
+                Await state.AssertLineTextAroundCaret($"        {committed}", "")
+            End Using
+        End Function
+
+        ' expressionAndCommitted explained:
+        ' expressionAndCommitted(0): the expression template before completion starts
+        ' expressionAndCommitted(1): the committed template text after completion
+        ' template markers:
+        ' ASYNC: if makeAsync, the modifier is removed, so the completion provider can add it
+        ' $$ in expressionAndCommitted(0): the cursor position
+        ' $$ in expressionAndCommitted(1): the cursor position after commit. If appendConfigureAwait,
+        ' ConfigureAwait(false) is expected to be inserted here and the cursor is after the ConfigureAwait(false)
+        <WpfTheory, CombinatorialData>
+        Public Async Function DotAwaitCompletionAddsAwaitInFrontOfExpressionInLambdas(<CombinatorialValues(
+                {
+                "await Task.Run(ASYNC () => Task.CompletedTask.$$",
```

Comment:
what's the reason for the capitaized ASYNC?

Reply:
It is described in the comment above the test:
```VB
        ' expressionAndCommitted explained:
        ' expressionAndCommitted(0): the expression template before completion starts
        ' expressionAndCommitted(1): the committed template text after completion
        ' template markers:
        ' ASYNC: if makeAsync, the modifier is removed, so the completion provider can add it
        ' $$ in expressionAndCommitted(0): the cursor position
        ' $$ in expressionAndCommitted(1): the cursor position after commit. If appendConfigureAwait,
        ' ConfigureAwait(false) is expected to be inserted here and the cursor is after the ConfigureAwait(false)
```

Reply:
please don't create dialects.  if you want to mark a special place in the code, just use $$ or[||] which the test harnesses already support.

Reply:
Resolving this discussion in favor of https://github.com/dotnet/roslyn/pull/55817#discussion_r704611544